src/DETHRACE/common/finteray.c

```diff
@@ -839,133 +833,113 @@ int ModelPickBox(br_actor* actor, tBounds* bnds, br_model* model, br_material* m
         return 0;
     }
     for (group = 0; prepared->ngroups > group; group++) {
-        fp = prepared->groups[group].faces;
         for (f = 0; f < prepared->groups[group].nfaces; f++) {
+            fp = &prepared->groups[group].faces[f];
             v1 = fp->vertices[0];
-            a.v[0] = prepared->groups[group].vertices[v1].p.v[0] - bnds->box_centre.v[0];
```

Comment:
it breaks my brain looking at this bit, I'm going to skip past and assume its all good! :). Nice one!

Reply:
I'm fixing the conflicts atm.
It's okay if you need more time to check everything.

Reply:
all good, happy to merge as-is. I think these days we have enough history functional behavior in PRs that it is relatively easy to step back in time if we see strange behavior and tie it to a particular PR 👍